crates/validation/src/bundle/mod.rs

```diff
@@ -0,0 +1,91 @@
+use std::{fmt::Debug, pin::Pin, sync::Arc};
+
+use alloy::{primitives::Address, sol_types::SolCall};
+use angstrom_types::contract_payloads::angstrom::{AngstromBundle, BundleGasDetails};
+use angstrom_utils::key_split_threadpool::KeySplitThreadpool;
+use eyre::eyre;
+use futures::{Future, FutureExt};
+use pade::PadeEncode;
+use revm::primitives::{EnvWithHandlerCfg, TxKind};
+use tokio::runtime::Handle;
+
+use crate::common::TokenPriceGenerator;
+
+pub mod validator;
+pub use validator::*;
+
+pub struct BundleValidator<DB> {
+    db:               Arc<DB>,
+    angstrom_address: Address,
+    /// the address associated with this node.
+    /// this will ensure the  node has access and the simulation can pass
+    node_address:     Address
+}
+
+impl<DB> BundleValidator<DB>
+where
+    DB: Unpin + Clone + 'static + reth_provider::BlockNumReader + revm::DatabaseRef + Send + Sync,
+    <DB as revm::DatabaseRef>::Error: Send + Sync + Debug
+{
+    pub fn new(db: Arc<DB>, angstrom_address: Address, node_address: Address) -> Self {
+        Self { db, angstrom_address, node_address }
+    }
+
+    pub fn simulate_bundle(
+        &self,
+        sender: tokio::sync::oneshot::Sender<eyre::Result<BundleGasDetails>>,
+        bundle: AngstromBundle,
+        price_gen: &TokenPriceGenerator,
+        thread_pool: &mut KeySplitThreadpool<
+            Address,
+            Pin<Box<dyn Future<Output = ()> + Send>>,
+            Handle
+        >
+    ) {
+        let node_address = self.node_address;
+        let angstrom_address = self.angstrom_address;
+        let db = self.db.clone();
+
+        let conversion_lookup = price_gen.generate_lookup_map();
+
+        thread_pool.spawn_raw(
+            async move {
+                let bundle = bundle.pade_encode();
+
+                {
+                    let mut evm = revm::Evm::builder()
```

Comment:
`{` ? Left out experiment?

Reply:
ah ye copied code from gas.rs. can delete since not dealing with lifetimed borrows